### PR TITLE
[Teacher][MBL-14813] View Syllabus

### PR DESCRIPTION
**Pull request by tamaskozmer**
Testing: Testing can be done by comparing with the student app. The syllabus should look the same as in the student app, with one difference. In the teacher app there aren't locked items in the syllabus events. Also clicking on a syllabus event, which is not an assignment is not working yet, will be implmented in a follow up ticket.

refs: MBL-14813
affects: Teacher
release note: Implemented option to view the course syllabus

### CODE REVIEW

File: apps/teacher/src/androidTest/java/com/instructure/teacher/ui/renderTests/SyllabusRenderTest.kt [hunk 1]
```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2019 - present Instructure, Inc.
```
Comment (Blackchai) [inline apps/teacher/src/androidTest/java/com/instructure/teacher/ui/renderTests/SyllabusRenderTest.kt:2]: Not a big deal, but copyright should be 2020

File: apps/teacher/src/androidTest/java/com/instructure/teacher/ui/pages/SyllabusPage.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
+/*
```
Comment (Blackchai) [inline apps/teacher/src/androidTest/java/com/instructure/teacher/ui/pages/SyllabusPage.kt:1]: copyright 2020

File: apps/teacher/src/androidTest/java/com/instructure/teacher/ui/SyllabusPageTest.kt [hunk 3]
```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2019 - present Instructure, Inc.
```
Comment (Blackchai) [inline apps/teacher/src/androidTest/java/com/instructure/teacher/ui/SyllabusPageTest.kt:2]: 2020

File: apps/teacher/src/androidTest/java/com/instructure/teacher/ui/renderTests/pages/SyllabusRenderPage.kt [hunk 4]
```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2019 - present Instructure, Inc.
```
Comment (Blackchai) [inline apps/teacher/src/androidTest/java/com/instructure/teacher/ui/renderTests/pages/SyllabusRenderPage.kt:2]: 2020, I'm guessing either the file settings or copy pasta issues :)

File: apps/teacher/src/main/java/com/instructure/teacher/features/syllabus/SyllabusEffectHandler.kt [hunk 5]
```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2019 - present Instructure, Inc.
```
Comment (Blackchai) [inline apps/teacher/src/main/java/com/instructure/teacher/features/syllabus/SyllabusEffectHandler.kt:2]: copyright

File: apps/teacher/src/main/java/com/instructure/teacher/features/syllabus/SyllabusPresenter.kt [hunk 6]
```diff
@@ -0,0 +1,86 @@
+package com.instructure.teacher.features.syllabus
```
Comment (Blackchai) [inline apps/teacher/src/main/java/com/instructure/teacher/features/syllabus/SyllabusPresenter.kt:1]: Missing license

File: apps/teacher/src/main/java/com/instructure/teacher/features/syllabus/SyllabusModels.kt [hunk 7]
```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2019 - present Instructure, Inc.
```
Comment (Blackchai) [inline apps/teacher/src/main/java/com/instructure/teacher/features/syllabus/SyllabusModels.kt:2]: 2020

File: apps/teacher/src/main/java/com/instructure/teacher/features/syllabus/SyllabusUpdate.kt [hunk 8]
```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2019 - present Instructure, Inc.
```
Comment (Blackchai) [inline apps/teacher/src/main/java/com/instructure/teacher/features/syllabus/SyllabusUpdate.kt:2]: 2020

File: apps/teacher/src/main/java/com/instructure/teacher/features/syllabus/ui/SyllabusEventsAdapter.kt [hunk 9]
```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2019 - present Instructure, Inc.
```
Comment (Blackchai) [inline apps/teacher/src/main/java/com/instructure/teacher/features/syllabus/ui/SyllabusEventsAdapter.kt:2]: 2020 :)

File: apps/teacher/src/main/java/com/instructure/teacher/features/syllabus/ui/SyllabusFragment.kt [hunk 10]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2019 - present Instructure, Inc.
```
Comment (Blackchai) [inline apps/teacher/src/main/java/com/instructure/teacher/features/syllabus/ui/SyllabusFragment.kt:2]: 2020, just marking all of these so we can check that they get changed, really no big deal

File: apps/teacher/src/main/java/com/instructure/teacher/features/syllabus/ui/SyllabusTabAdapter.kt [hunk 11]
```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2019 - present Instructure, Inc.
```
Comment (Blackchai) [inline apps/teacher/src/main/java/com/instructure/teacher/features/syllabus/ui/SyllabusTabAdapter.kt:2]: 2020

File: apps/teacher/src/main/java/com/instructure/teacher/features/syllabus/ui/SyllabusView.kt [hunk 12]
```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) 2019 - present Instructure, Inc.
```
Comment (Blackchai) [inline apps/teacher/src/main/java/com/instructure/teacher/features/syllabus/ui/SyllabusView.kt:2]: 2020

File: apps/teacher/src/main/java/com/instructure/teacher/features/syllabus/ui/SyllabusViewState.kt [hunk 13]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2019 - present Instructure, Inc.
+ *
```
Comment (Blackchai) [inline apps/teacher/src/main/java/com/instructure/teacher/features/syllabus/ui/SyllabusViewState.kt:3]: 2020

File: apps/teacher/src/main/res/layout/fragment_syllabus.xml [hunk 14]
```diff
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="utf-8"?><!--
+  ~ Copyright (C) 2019 - present Instructure, Inc.
```
Comment (Blackchai) [inline apps/teacher/src/main/res/layout/fragment_syllabus.xml:2]: 2020

File: apps/teacher/src/main/res/layout/fragment_syllabus_events.xml [hunk 15]
```diff
@@ -0,0 +1,78 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright (C) 2019 - present Instructure, Inc.
```
Comment (Blackchai) [inline apps/teacher/src/main/res/layout/fragment_syllabus_events.xml:3]: 2020

File: apps/teacher/src/main/res/layout/fragment_syllabus_webview.xml [hunk 16]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright (C) 2019 - present Instructure, Inc.
```
Comment (Blackchai) [inline apps/teacher/src/main/res/layout/fragment_syllabus_webview.xml:3]: 2020

File: apps/teacher/src/main/res/layout/viewholder_syllabus_item.xml [hunk 17]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="utf-8"?><!--
+  ~ Copyright (C) 2019 - present Instructure, Inc.
```
Comment (Blackchai) [inline apps/teacher/src/main/res/layout/viewholder_syllabus_item.xml:2]: 2020

File: apps/teacher/src/test/java/com/instructure/teacher/features/syllabus/SyllabusEffectHandlerTest.kt [hunk 18]
```diff
@@ -0,0 +1,278 @@
+package com.instructure.teacher.features.syllabus
```
Comment (Blackchai) [inline apps/teacher/src/test/java/com/instructure/teacher/features/syllabus/SyllabusEffectHandlerTest.kt:1]: missing license

File: apps/teacher/src/test/java/com/instructure/teacher/features/syllabus/SyllabusPresenterTest.kt [hunk 19]
```diff
@@ -0,0 +1,194 @@
+package com.instructure.teacher.features.syllabus
```
Comment (Blackchai) [inline apps/teacher/src/test/java/com/instructure/teacher/features/syllabus/SyllabusPresenterTest.kt:1]: license

File: libs/pandautils/src/main/java/com/instructure/pandautils/utils/DateExtensions.kt [hunk 20]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2019 - present Instructure, Inc.
```
Comment (Blackchai) [inline libs/pandautils/src/main/java/com/instructure/pandautils/utils/DateExtensions.kt:2]: 2020

**Review by Blackchai — CHANGES_REQUESTED** [review]
This is looking great. I'm going to do testing now, the only changes I'm requesting are the super minor licensing ones.

**Comment by tamaskozmer** [comment]
I don't know why some classes does not have the license, because I set it up as an automatic header. Maybe if I create a Kotlin file instead of a Kotlin class it doesn't add the licence, I will look into it. I just copied one licence header from a class, and added it to Android Studio, but haven't checked the date on it.

**Comment by Blackchai** [comment]
Something I just noticed this morning, I don't see a remote config param for this feature on firebase. Do you need help creating one?

**Review by Blackchai — APPROVED** [review]
LGTM!

**Pull request merged**